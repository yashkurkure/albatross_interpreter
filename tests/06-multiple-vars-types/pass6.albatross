var i1 int := 0;
var s1 string := "hello";
var i2 int := 0;
var s2 string := "world";
var i3 int := 0;
var s3 string := "cs473";
var i4 int := 0;

s1 := s2;
s2 := "";

i1 := 400;
i2 := i2 + 70;
i3 := i3 + 3;
return i3 + i4;
