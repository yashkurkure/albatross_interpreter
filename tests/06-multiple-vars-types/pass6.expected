Variable declared "i1" type int
Variable declared "s1" type string
Variable declared "i2" type int
Variable declared "s2" type string
Variable declared "i3" type int
Variable declared "s3" type string
Variable declared "i4" type int
Variable read "s2" type string
Variable written "s1" type string
Variable written "s2" type string
Variable written "i1" type int
Variable read "i2" type int
Variable written "i2" type int
Variable read "i3" type int
Variable written "i3" type int
Variable read "i3" type int
Variable read "i4" type int
